### DISCUSSION TITLE
Weird Red Screen of Death

### DISCUSSION THREAD

**Discussion by rlovell3** [mylinuxforwork/dotfiles#212] [General]
I've been installing arch, and immediately installing your ml4w dotfilles onto numerous machines the past few weeks.  All flawless until this one.  My most recent is the first to use 2.9.6.  I've got Hyperlock Timeout @ 1200.  DPMS Timeout at 1260.  Suspend Timeout at 0.  At 20 minutes, the circle appears indicating screen lock.  At 41 minutes the screen goes totally black, followed a few seconds later by a completely RED SCREEN.  There is no key binding to escape out of it.  The only solution is to manually hold down the power button and force the machine to shut down.  This behavior was replicated accidentally simply by clicking the "Suspend" button on the main Power/Hibernate/Reboot/Suspend menu-window.  What might be sending a "Suspend" signal, and how would you go about diagnosing this? I'm not sure if it is a graphic driver issue or somehow some laptop power-saving service got activated.  Just not sure about this one.  Any ideas? Thank you, and thanks for flipping my switch toward Arch and Wayland/Hyprland.  Amazing path forward.

**Comment by ajboni** [comment]
I had this issue with 'vanilla' hyprlnd and it turned out to be something related to `dpms` and `swaylock` (I was coming from sway) I did not managed to fix it as it was fixed for me oob when I installed ml4w. But it was definitely something that was kicking in after the timeout period for suspend.

Instead of rebooting you can try `ctrl`+`alt`+`f3` and do `killall Hyprland`

**Reply by rlovell3** [reply]
That is excellent info. Your command works to get me into a shell. It's only a matter of time now before I figure out how to stop seeing red.  DPMS seems to be the culprit.  Will post my notes here when I find a fix for this.

**Comment by rlovell3** [comment]
This is a follow-up for anyone who bumps into a red screen of death.  In my case, it was specific to a single monitor.  That monitor could not properly handle DPMS, even though the monitor had settings for sleep and blank.  It was a pain to figure out. Here are the steps:  

### Red Screen of Death  

#### Determine what your monitor is doing with respect to dpms:

Find your monitor:  
`for dev in /sys/class/drm/*/status; do echo "$dev:"; cat "$dev"; echo ""; done`  

Once you locate the /sys/class/drm/ monitor, output its data to a text file:  
`sudo cat /sys/class/drm/cardX-Y/edid > ~/monitor.edid`  replace cardX-Y with your monitor. 

Read the monitor.edid file with edid-decoder
`sudo paru -S edid-decode`
edid-decode monitor.edid

Look for your dpms settings. If you only have "Off", your monitor can't handle dpms.
Set dpms_timeout=0 in your ml4w settings.

Install swayidle and swaylock
paru -S swayidle
paru -S swaylock

edit ~/.config/hypr/conf/custom.conf: 
Edit custom.conf to use swayidle and swaylock.  Be sure wo map to the exact executables, i.e.: `which swayidle` and `which swaylock`, and use those paths in your command below:  
 
```exec=/usr/bin/swayidle timeout 300 '/usr/bin/swaylock -f -c 000000' before-sleep '/usr/bin/swaylock -f -c 000000'  
```
That command uses 300 seconds to go to a locked black screen.  Change the number of seconds to suit your own preferences.

**Reply by rlovell3** [reply]
Apparently there were too many quotation marks in the actual command to add to custom.conf, so here it is again, but without the encapsulating quotation marks:  

exec=/usr/bin/swayidle timeout 300 '/usr/bin/swaylock -f -c 000000' before-sleep '/usr/bin/swaylock -f -c 000000'
